protos/sigstore_trustroot.proto

```diff
@@ -148,9 +148,11 @@ message TrustedRoot {
 // signing. In particular, it primarily contains service URLs that a Sigstore
 // signer may need to connect to for the online aspects of signing.
 message SigningConfig {
-        // MUST be application/vnd.dev.sigstore.signingconfig.v0.1+json
+        // MUST be application/vnd.dev.sigstore.signingconfig.v0.1+json or
+        // application/vnd.dev.sigstore.signingconfig.v0.2+json
```

Comment:
This seems a bit unclear: after this PR this message documents `v0.2` only but that's not said anywhere. Do we allow v0.1 to be used in future as well?

Bundle uses this language:
>         // MUST be application/vnd.dev.sigstore.bundle.v0.3+json when encoded as JSON.
>         // Clients must to be able to accept media type using the previously defined formats:
>         // [list follows]

Of course situation here is a bit different: we should document 
* what we expect the instance to serve 
  * I'd like to say "MUST serve v0.2" but obviously there's a migration period here
* what we expect clients to handle
  * MUST handle v0.2, SHOULD handle v0.1

Reply:
Updated to serve only v0.2, and say clients can optionally handle v0.1. In practice, I assume all will drop support.